test/test_art_key.cpp

```
@@ -0,0 +1,91 @@
+  EXPECT_EQ(static_cast<std::byte>(0x06), ikey[5]);
+  EXPECT_EQ(static_cast<std::byte>(0x07), ikey[6]);
+  EXPECT_EQ(static_cast<std::byte>(0x08), ikey[7]);
+}
+
+// checks shift_right
+TYPED_TEST(ARTKeyTest, basic_art_key_C0010) {
+  const std::uint64_t ekey1 = 0x0102030405060708;            // external key
+  const std::uint64_t ekey2 = 0x0304050607080000;            // external key
+  const TypeParam ikey1(ekey1);                              // encode
+  const TypeParam ikey2(ekey2);                              // encode
+  const std::uint64_t akey2 = decode(ikey2.get_key_view());  // round trip
```

Cppcheck (2.10-2)

[error] There is an unknown macro here somewhere. Configuration is required. If UNODB_DETAIL_DISABLE_CLANG_WARNING is a macro then please configure it.

(unknownMacro)